solidity/contracts/token/libs/FeeTokenRouter.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {TokenRouter} from "./TokenRouter.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+struct Quotes {
+    address token;
+    uint256 amount;
+}
+
+/**
+ * @title Warp Route that charges non-native (ERC20) fees in addition to the native IGP fee.
+ * @author Abacus Works
+ */
+abstract contract FeeTokenRouter is TokenRouter {
```

Comment:
This should probably implement the `IValueTransferBridge` interface?

Reply:
Done